readthedocs/rtd_tests/tests/test_build_storage.py

```diff
@@ -27,6 +46,39 @@ def test_copy_directory(self):
         self.assertTrue(self.storage.exists('files/api.fjson'))
         self.assertTrue(self.storage.exists('files/api/index.html'))
 
+    def test_sync_directory(self):
+        tmp_files_dir = os.path.join(tempfile.mkdtemp(), 'files')
+        shutil.copytree(files_dir, tmp_files_dir)
+        storage_dir = 'files'
+
+        tree = [
+            ('api', ['index.html']),
+            'api.fjson',
+            'conf.py',
+            'test.html',
+        ]
+        self.storage.sync_directory(tmp_files_dir, storage_dir)
+        self.assertFileTree(storage_dir, tree)
+
+        tree = [
+            ('api', ['index.html']),
+            'conf.py',
+            'test.html',
+        ]
+        os.remove(os.path.join(tmp_files_dir, 'api.fjson'))
+        self.storage.sync_directory(tmp_files_dir, storage_dir)
+        self.assertFileTree(storage_dir, tree)
+
+        tree = [
+            # Cloud storage generally doesn't consider empty directories to exist
+            ('api', []),
```

Comment:
I was about to fix this to delete all dirs on `FileSystemStorage`, but looks like we expect that and probably isn't a big deal.

https://github.com/readthedocs/readthedocs.org/blob/43463b4549e766891c1b5a397a6b286faa76ece3/readthedocs/rtd_tests/tests/test_build_storage.py#L39-L41